legate/numpy/runtime.py

```diff
@@ -585,33 +584,20 @@ def find_or_create_key_partition(self):
             if self.transform:
                 lo = self.transform.apply(lo)
                 hi = self.transform.apply(hi)
-            # Compute the lower bound tile and upper bound tile
+            # Compute the lower bound and upper bound tile on the root space
             assert len(lo) == len(root_key.tile_shape)
             color_lo = tuple(map(lambda x, y: x // y, lo, root_key.tile_shape))
             color_hi = tuple(map(lambda x, y: x // y, hi, root_key.tile_shape))
             color_tile = root_key.tile_shape
             if self.transform:
-                # Check to see if this transform is invertible
-                # If it is then we'll reuse the key partition of the
-                # root in order to guide how we do the partitioning
-                # for this view to maximimize locality. If the transform
-                # is not invertible then we'll fall back to doing the
-                # standard mapping of the index space
-                invertible = True
```

Comment:
The original check was not strict enough. For example, the transform `x = np.array([[0,0,1], [1,0,0], [0,1,0], [0,0,0]])` would pass the check, but `x @ x.T != I`.

Reply:
I think that was intentional. I was trying to handle the cases where the user added or remove dimensions for broadcasting, e.g. with `np.newaxis`. We should still be able to invert such mappings, even if it is not in the strict linear-algebra sense invertible.

Reply:
Makes sense, I will confirm that everything still works with added/removed dimensions and relax the check.